### ISSUE TITLE
Problem about the implementation of lmpt/csel.py.

### ISSUE THREAD

**Issue by Wang-zuozheng** [richard-peng-xia/LMPT#5]
Some of the code is different from the paper.
1. Computing $n_i$
In the paper $n_i$ denotes the number of training examples that contain class i. In the code, neg_class_ferq is loaded (the number of negative samples).

2. Computing μ
In the paper，
![image](https://github.com/richard-peng-xia/LMPT/assets/45785149/7c00e360-0ae0-46a4-8dd7-e6b5002299c1)
In the code ![image](https://github.com/richard-peng-xia/LMPT/assets/45785149/8258ef34-fcd8-4cc2-9dc8-b2e7edf1c143)
There is an addition softmax operation.

Can not execute.
![image](https://github.com/richard-peng-xia/LMPT/assets/45785149/daa9a38b-cd26-447b-8ee4-dd01d62344bf)
because self.class_counts  has value 0.

I wondering if there are any issues with the implementation of the CSE loss function? thanks!

**Comment by richard-peng-xia** [comment]
Hi @Wang-zuozheng,

Thanks for your interest in LMPT!

1.  Thank you for reminding me. In fact, it should be written as `class_weights = sfm(torch.from_numpy(np.asarray(mmcv.load(freq_file)['class_freq'])).to(torch.float32).cuda())`, where I forgot to update the code. Details can be found [here](https://github.com/wutong16/DistributionBalancedLoss/blob/master/mllt/datasets/custom.py#L320).

2. It is not an addition softmax operation. This line of code corresponds to Eq. 5, which is the re-weight operation.

3. This error should not happen as there are corresponding images for each category, although the number varies.

I hope it may help you. Please feel free to ask if there is any further problem.

Thank you very much!
